ark/utils/plot_utils.py

```diff
@@ -356,3 +357,106 @@ def create_overlay(fov, segmentation_dir, data_dir,
         rescaled[alternate_contour_mask > 0, 1:] = 0
 
     return rescaled
+
+
+def create_mantis_project(fovs: List[str], mantis_project_path: Union[str, pathlib.Path],
+                          img_data_path: Union[str, pathlib.Path],
+                          mask_output_dir: Union[str, pathlib.Path],
+                          mapping_path: Union[str, pathlib.Path],
+                          seg_dir: Union[str, pathlib.Path],
+                          mask_suffix: str = "_mask", img_sub_folder: str = "normalized"):
+    """Creates a mantis project directory so that it can be opened by the mantis viewer.
+    Copies fovs, segmentation files, masks, and mapping csv's into a new directory structure.
+    Here is how the contents of the mantis project folder will look like.
+
+    ```{code-block} sh
+    mantis_project
+    ├── fov0
+    │   ├── cell_segmentation.tiff
+    │   ├── chan0.tiff
+    │   ├── chan1.tiff
+    │   ├── chan2.tiff
+    │   ├── ...
+    │   ├── population_mask.csv
+    │   └── population_mask.tiff
+    └── fov1
+    │   ├── cell_segmentation.tiff
+    │   ├── chan0.tiff
+    │   ├── chan1.tiff
+    │   ├── chan2.tiff
+    │   ├── ...
+    │   ├── population_mask.csv
+    │   └── population_mask.tiff
+    └── ...
+    ```
+
+    Args:
+        fovs (List[str]):
+            A list of FOVs to create a Mantis Project for.
+        mantis_project_path (Union[str, pathlib.Path]):
+            The folder where the mantis project will be created.
+        img_data_path (Union[str, pathlib.Path]):
+            The location of the all the fovs you wish to create a project from.
+        mask_output_dir (Union[str, pathlib.Path]):
+            The folder containing all the masks of the fovs.
+        mapping_path (Union[str, pathlib.Path]):
+            The location of the mapping path file.
+        seg_dir (Union[str, pathlib.Path]):
+            The location of the segmentation directory for the fovs.
+        mask_suffix (str, optional):
+            The suffix used to find the mask tiffs. Defaults to "_mask".
+        img_sub_folder (str, optional):
+            The subfolder where the channels exist within the `img_data_path`.
+            Defaults to "normalized".
+    """
+
+    if not os.path.exists(mantis_project_path):
+        os.makedirs(mantis_project_path)
+
+    # create key from cluster number to cluster name
+    map_df = pd.read_csv(mapping_path)
```

Comment:
Can you change this to accept an already loaded dataframe, instead of loading one from a path?

Reply:
Will we ever need to read it in from a `csv` file? Or should it be assumed to always be from a Pandas DataFrame that's already loaded in memory?

Reply:
In the pixel clustering notebooks it's a saved file, like in the current setup. However, in other notebooks it will already be in memory. Seems simplest to me to add a line of code that loads the saved CSVs for the clustering notebooks, rather than saving a file in the other notebooks. It's also possible to make it handle both a path and a loaded df, but that seems a bit clunky. It's up to you

Reply:
Do you mean once this is pushed, we adjust the path to always point to the Mantis section in the README.md which is on the master branch?

So something like `https://github.com/angelolab/ark-analysis/blob/master/README.md#mantis-viewer` instead of `../README.md#mantis-viewer`?

Reply:
If we merge this into master as is, and the branch is deleted, won't the link break?

Reply:
I do not believe so, as it is a relative link. As long as the `README.md` is outside of the templates folder, it should find it and get the `mantis-viewer` category.

Reply:
I'm an idiot, I was looking at the link it was taking me to, not the actual link in the document. Yeah I agreed, this fine.